src/sidebar/services/oauth-auth.js

```diff
@@ -205,50 +207,52 @@ export default function auth(
 
     const origToken = tokenInfoPromise;
 
-    return tokenInfoPromise.then(token => {
-      if (!token) {
-        // No token available. User will need to log in.
-        return null;
-      }
+    return /** @type {Promise<TokenInfo|null>} */ (tokenInfoPromise).then(
```

Comment:
There are no changes to the body of the `then` callback. It was just necessary in this context to give TS a bit of help to understand the type of `tokenInfoPromise` in this slightly complex method. It is also useful for the human reader IMO.